Python/ds/Circular_LinkedList.py

```diff
@@ -0,0 +1,161 @@
+        self.tail = node
+        node.next = self.head
+        self.size = self.size+1
+        print("Value",node.value,"adding at the END of list")
+        self.printAll()
+        
+
+    # Insert node given as argument to the start of the list
+    def prepend(self,node):
+        '''insert node given as argument to the start of the list'''
+        node.next = self.head
+        self.head = node
+        self.tail.next = self.head
+        self.size = self.size+1
+        print("Value",node.value,"adding at the START of list")
+        self.printAll()
+
+    # Prints the no of nodes or size of linked list
+    def getSize(self):
+        '''Prints the no of nodes or size of linked list'''
+        print("Total No of Nodes :",end=" ")
+        print(self.size)
+
+
+    # Print all the nodes of list
+    def printAll(self):
+        '''Print all the nodes of list'''
+        node = self.head
+
+        print("Your Current List: ")
+
+        while node.next!=self.head:
+
+            print(node.value,end='')
+            print("--->",end='')
+            node = node.next
+
+        else:
+            print(node.value)
+
+
+    # Insert node at position passed in argument
+    def insert(self,n,index):
+        '''Insert node at position passed in argument
+
+        usage : insert(Node,position)
+
+        '''
+        i = 0
+        node = self.head
+        while node:
+            if i==index-1:
+                nextNode = node.next
+                node.next = n
+                n.next = nextNode
+                self.size = self.size+1
+                break
+            else:
+                i+=1
+                node=node.next
+        else:
+            print("Given index is not in proper range")
+
+        print("Inserting",n.value,"at index",index)
+        
+        self.printAll()
+
+    # Remove node at position passed in argument
+    def remove(self,data):
+        '''Remove node at position passed in argument
+
+        usage : remove(value)
+
+        '''
+        node = self.head
+        while node:
+            if node.next.value == data:
+                nextNode = node.next.next
+                node.next = nextNode
+                self.size = self.size-1
+                break
+            else:
+                node = node.next
+
+        else:
+            print("Value not Found")
+
+        print("Removing",data,"from the list")
+        
+        self.printAll()
+
+
+# Class for creation of nodes in Linked List
+class Node:
+    def __init__(self,data):
+        self.value = data
+        self.next = None
+
+if __name__ == '__main__':
+
+    # Taking Input from User
+    user_input = list(map(int,input("Enter values to be added in List \n").split()))
+
+    # Setting First Value as HEAD
+    l  = circular_llist(Node(user_input[0]))
+
+    # Appending values in list
+    for inputs in range(1,len(user_input)):
+        l.create(Node(user_input[inputs]))
+
+    l.printAll()
+    l.append(Node(10))
```

Comment:
Hey @Shishir198 please take these inputs from the user i.e the value to be appended at the start, value to be appended at the end etc.

Reply:
Okay mam. Do you mean for every function I had to ask user for input.

Reply:
Yes @Shishir198, please ask user for the input for each operation instead of hardcoding it.